src/main/java/codeu/controller/TagServlet.java

```diff
@@ -90,25 +90,33 @@ void setTagStore(TagStore tagStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) 
       throws IOException, ServletException {
+
+    String username = (String) request.getSession().getAttribute("user");
+    if (username == null) {
+      // User is not logged in, don't let them see a profile
+      System.out.println("Please login before viewing tagged conversations");
+      response.sendRedirect("/login");
+      return;
+    }
+
     String requestUrl = request.getRequestURI();
     Matcher m = Pattern.compile("^/tag/(.+)").matcher(requestUrl);
     if (!m.find()) {
       System.out.println("wrong URL pattern: " + requestUrl);
       return; // return error 
     }
-
     String extractedTag = m.group(1);
-    Tag thisTag = tagStore.getTag(extractedTag);
-
-    List<UUID> conversationIDs = thisTag.getConversations();
-
     List<String> conversationTitles = new ArrayList<>();
-    for (UUID id : conversationIDs) {
-      Conversation convo = conversationStore.getConversationWithID(id);
-      conversationTitles.add(convo.getTitle());
+
+    // Sets request attribute to a list of Conversation titles that are associated with the Tag
+    for (Tag tag : tagStore.getAllTags()) {
+      if (tag.getTag().equals(extractedTag)) {
```

Comment:
I think we overwrote the `Tag.equals()` method to only check for equality based on the tag string, so maybe just use that directly. I think I saw this equality check in several other places in #72 as well.

Reply:
`extractedTag` is a String parsed from the URL, so I'm using that to compare, instead of two `Tag` objects